### PR TITLE
audit: update to 3.1.5

### PR DESCRIPTION
**Pull request by zpc0**
- Remove upstreamed patch. : 0001-Implicit-builtin-functions.patch
- Fix package license info.
- Use git mirror instead of GitHub generated tarball.

Compile tested: mvebu/FG-50E (host/target)

### CODE REVIEW

**Comment by Neustradamus** [comment]
@zpc0: Good PR! :)

File: package/utils/audit/Makefile [hunk 1]
```diff
@@ -6,14 +6,18 @@
...
-PKG_RELEASE:=2
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=https://github.com/linux-audit/audit-userspace/archive/refs/tags/v$(PKG_VERSION).tar.gz?
-PKG_HASH:=aec501760acd13ebbe00e78b9b59f795d16a430b1d673628e346cd18905c594b
+PKG_VERSION:=3.1.5
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/linux-audit/$(PKG_NAME).git
+PKG_SOURCE_DATE:=2024-07-29
```
Comment (hauke) [inline package/utils/audit/Makefile:14]: Please remove the `PKG_SOURCE_DATE` to use the version number in the package only.

File: package/utils/audit/Makefile [hunk 2]
```diff
@@ -6,14 +6,18 @@
...
+PKG_SOURCE_VERSION:=f9f228e8661527fb0a064530b2dc22b78fe21c40
```
Comment (neheb) [inline package/utils/audit/Makefile:15]: this should be

```
PKG_SOURCE_VERSION:=v$(PKG_VERSION)
```

**Comment by zpc0** [comment]
@hauke @neheb Thank you for review! I fixed both.

**Pull request merged**